Sources/Timer.swift

```diff
@@ -17,9 +17,16 @@ struct Timer {
     init(startingAt startingTimeInMilliseconds: Double = 0.0) {
         var startTime = timeval()
         gettimeofday(&startTime, nil)
-        if !startingTimeInMilliseconds.isZero {
-            startTime.tv_usec += type(of: startTime.tv_usec).init(startingTimeInMilliseconds * 1000)
+
+        let seconds = (startingTimeInMilliseconds / 1000).rounded(.down)
+        let milliseconds = startingTimeInMilliseconds.truncatingRemainder(dividingBy: 1000)
+        if !seconds.isZero {
```

Comment:
As an aside: it doesn't seem necessary to check whether `seconds.isZero`? We would just `0 += 0` in that case – juckt?